src/app/task/task-list-item/task-list-item.component.ts

```diff
@@ -108,58 +109,51 @@ export class TaskListItemComponent implements OnChanges {
     return types[indexOfNextStatus];
   }
 
-  changeStatus(): TaskStatusType | undefined {
+  changeStatus(): void {
     if (!this.editable) return;
     let state = this.getNextStatusIndex();
-    return this.taskStatus = state;
+    this.taskStatus = state;
   }
 
   getBackgroundColor(): string | undefined {
     let result: string | undefined;
     switch (this.taskStatus) {
       case StatusEnum.toDo:
-        result = 'inherit';
-        break;
+        return result = 'inherit';
       case StatusEnum.inProgress:
-        result = 'violet';
-        break;
+        return result = 'violet';
       case StatusEnum.done:
-        result = 'green';
-        break;
+        return result = 'green';
     }
     return result;
   }
 
-  get priorities() {
-    const arr = Object.entries(PriorityEnum);
-    return arr.map(([title, color]) => ({ title, color }));
+  get priorities(): TaskPriorityInterface[] {
```

Comment:
there is no need for this method if you define priorities list as an array TaskPriorityInterface[] that mentioned in comment below